Revit_Core_Engine/Create/Sheet.cs

```diff
@@ -0,0 +1,83 @@
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using BH.oM.Base.Attributes;
+using System.ComponentModel;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /****               Public Methods              ****/
+        /***************************************************/
+
+        [Description("Creates and returns a new Floor Plan view in the current Revit file.")]
+        [Input("document", "The current Revit document to be processed.")]
+        [Input("sheetName", "Name of the new sheet.")]
+        [Input("sheetNumber", "Number of the new sheet.")]
+        [Input("viewTemplateId", "The Title Block Id to be applied to the sheet.")]
+        [Output("viewSheet", "The new sheet.")]
+        public static ViewSheet Sheet(this Document document, string sheetName, string sheetNumber, ElementId titleBlockId)
+        {
+            ViewSheet result = ViewSheet.Create(document, titleBlockId);
+
+            if (!string.IsNullOrEmpty(sheetName))
+            {
+                result.Name = sheetName;
+            }
+
+            if (!string.IsNullOrEmpty(sheetNumber))
+            {
+                int number = 0;
+
+                while (true)
+                {
+                    try
+                    {
+                        if (number != 0)
+                        {
+                            result.SheetNumber = $"{sheetNumber} ({number})";
+                            BH.Engine.Base.Compute.RecordWarning("There is already a sheet named '" + sheetNumber + "'." + " It has been named '" + result.SheetNumber + "' instead.");
+                            break;
+                        }
+
+                        result.SheetNumber = sheetNumber;
+                        break;
+                    }
+                    catch
+                    {
+                        number++;
+                    }
+                }
```

Comment:
I often feel nervous when seeing `while (true)` 😅 and also avoid `try catch` whenever I can.

In this case, it can `RecordWarning` multiple times if `number` has to increment a few times to make a unique sheet number. This will give a longer and less clear error log.

What I would do is create a method just for checking if a sheet number already exists (`IsExistingSheetNumber`) and then do something like this:
```c#
int number = 0;
string uniqueNumber = sheetNumber;

while (uniqueNumber.IsExistingSheetNumber())
{
    number++;
    uniqueNumber = $"{sheetNumber} ({number})";
}

result.SheetNumber = uniqueNumber;
if (uniqueNumber != sheetNumber)
{
    BH.Engine.Base.Compute.RecordWarning($"There is already a sheet named '{sheetNumber}'. It has been named '{uniqueNumber}' instead.");
}
```